lib/presentation/core/app_widget.dart

```diff
@@ -29,6 +29,12 @@ class AppWidget extends StatelessWidget {
         listener: (context, state) {
           BlocProvider.of<ProfileBloc>(context).add(GetUserProfile());
           BlocProvider.of<ProfileTabBloc>(context).add(FetchProfileTabInfo());
+
+          state.whenOrNull(
+            unauthenticated: () {
+              _appRouter.replaceAll([const UnauthenticatedRoute()]);
+            },
+          );
```

Comment:
These two Events should be called only when authenticated, at least I presume since they would fail if no User.